### PR TITLE
Simplifying the EOS and Transport structures

### PR DESCRIPTION
**Pull request by marchdf**
Curious about what people think of this idea...
- consolidates logic on EOS switches
- compiles in all the EOS (even though we can only pick which one to use at compile time)
- consolidates common EOS functions into one header and cpp file.
- removed all fortran EOS (does anyone still use those?)

The downside is that it requires a bit of rejigging of the make system (eg `Eos_dir` -> `Eos_Model` in the `GNUMakefile`)

If people like this, I might do it for transport too. 

@hariswaran @jrood-nrel @esclapez @drummerdoc, thoughts?

### CODE REVIEW

**Comment by drummerdoc** [comment]
Indifferent.  If you can get it to pass all the tests, and if you like it better this way for some reason, I'm good with it

**Comment by esclapez** [comment]
Same here. I don't think anyone is still using Fortran. I'll probably wait until you've updated the transport to make the changes to LM make system.

**Review by jrood-nrel — APPROVED** [review]
This is good. This allows us to compile all the EOSes in at the same time and even though switching the EOS at runtime doesn't make much sense, it will allow us to catch more problems, and is one step closer to allowing us to use libraries inside Pele.

**Comment by marchdf** [comment]
Hey everyone, I think this is good to go once the tests pass. LMK what you think.

**Review by esclapez — APPROVED** [review]
I'm good with these changes. One step closer to having runtime selection !

File: Testing/Exec/TranEval_FORTRAN/GNUmakefile [hunk 1]
```diff
@@ -19,15 +19,15 @@ PELE_PHYSICS_HOME    := ../../..
 USE_F90_PP = TRUE
 
 ifeq ($(FUEGO_GAS), TRUE)
-  Eos_dir         = Fuego
```
Comment (baperry2) [inline Testing/Exec/TranEval_FORTRAN/GNUmakefile:22]: If the Fortran EOS support is being removed, shouldn't the Eval_FORTRAN test cases also be removed?

**Review by baperry2 — CHANGES_REQUESTED** [review]
Looks good. Much easier to follow code organization now. You should also be able to remove the Fortran test cases now.

File: Testing/Exec/TranEval_FORTRAN/GNUmakefile (hunk 1 above)
Comment (marchdf) [reply Testing/Exec/TranEval_FORTRAN/GNUmakefile:22]: Good catch @baperry2 . It is done.

**Review by baperry2 — APPROVED** [review]

**Pull request merged**